app/packages/core/src/components/Modal/use-looker.ts

```diff
@@ -0,0 +1,103 @@
+import * as fos from "@fiftyone/state";
+import React, { useEffect, useRef, useState } from "react";
+import { useErrorHandler } from "react-error-boundary";
+import { useRecoilValue } from "recoil";
+import { v4 as uuid } from "uuid";
+import { useClearSelectedLabels, useShowOverlays } from "./ModalLooker";
+import { useLookerOptionsUpdate } from "./hooks";
+import useKeyEvents from "./use-key-events";
+import { shortcutToHelpItems } from "./utils";
+
+const CLOSE = "close";
+
+function useLooker<L extends fos.Lookers>({
+  sample,
+}: {
+  sample: fos.ModalSample;
+}) {
+  const [id] = useState(() => uuid());
+  const initialRef = useRef<boolean>(true);
+  const ref = useRef<HTMLDivElement>(null);
+  const [reset, setReset] = useState(false);
+  const lookerOptions = fos.useLookerOptions(true);
+  const createLooker = fos.useCreateLooker(
+    true,
+    false,
+    lookerOptions,
+    undefined,
+    true
+  );
+  const selectedMediaField = useRecoilValue(fos.selectedMediaField(true));
+  const colorScheme = useRecoilValue(fos.colorScheme);
+  const looker = React.useMemo(() => {
+    reset;
+    selectedMediaField;
```

Comment:
I think I see what you're trying to do here, do you mind adding comments to clarify it?